### PR TITLE
Add template for release PR

### PR DESCRIPTION
**Pull request by BenjaminRueth**
Using a link with [query parameters](https://help.github.com/en/articles/about-automation-for-issues-and-pull-requests-with-query-parameters), we can use a PR template when a PR is created. This means, we should use the following link to use the proposed template: 

```
https://github.com/precice/precice/compare/new?template=release_pull_request_template.md
```

I cannot really test this, since (I assume) the template has to be provided on develop. I would suggest to first finalize the content of the new PR template, then merge it to develop and finally, we can make sure that the template shows up at the right places and can be used.

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE/release_pull_request_template.md [hunk 1]
```diff
@@ -0,0 +1,52 @@
+## Step by step guide
+* [ ] Merge master to develop (No commits after the release on master) @MakisH
```
Comment (MakisH) [inline .github/PULL_REQUEST_TEMPLATE/release_pull_request_template.md]: In the template we only want the steps. The people change all the time.

File: .github/PULL_REQUEST_TEMPLATE/release_pull_request_template.md [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+* [ ] Create branch `release-N` from develop. If needed, `git rebase develop`. @MakisH 
+* [ ] Open PR from `release-N` to `master` (use [this template](https://github.com/precice/precice/blob/add_PR_template/.github/PULL_REQUEST_TEMPLATE/release_pull_request_template.md)) @MakisH 
+* [ ] Look over [`CHANGELOG.md`](https://github.com/precice/precice/blob/develop/CHANGELOG.md) and add things if necessary and extract summary (all) @fsimonis 
+* [ ] Do regression tests using the release branch (specific revision) _list below :arrow_down:_ (all)
+* [ ] Bump version in CMakeLists.txt, SConstruct, and CHANGELOG (and Python bindings). @MakisH 
+* [ ] Draft message to mailing list @uekerman @MakisH 
+* [ ] Update documentation (all) @MakisH 
+* [ ] Fix potential problems in develop (all)
+* [ ] Merge develop to master @MakisH 
+* [ ] Tag release (on GitHub) and merge back to develop @MakisH
```
Comment (MakisH) [inline .github/PULL_REQUEST_TEMPLATE/release_pull_request_template.md]:
```suggestion
* [ ] Tag release (on GitHub, make sure to select the release branch as target) and merge back to develop
```

File: .github/PULL_REQUEST_TEMPLATE/release_pull_request_template.md [hunk 3]
```diff
@@ -0,0 +1,52 @@
+## Step by step guide
+* [ ] Merge master to develop (No commits after the release on master) @MakisH 
+* [ ] Create branch `release-N` from develop. If needed, `git rebase develop`. @MakisH 
+* [ ] Open PR from `release-N` to `master` (use [this template](https://github.com/precice/precice/blob/add_PR_template/.github/PULL_REQUEST_TEMPLATE/release_pull_request_template.md)) @MakisH 
+* [ ] Look over [`CHANGELOG.md`](https://github.com/precice/precice/blob/develop/CHANGELOG.md) and add things if necessary and extract summary (all) @fsimonis 
+* [ ] Do regression tests using the release branch (specific revision) _list below :arrow_down:_ (all)
+* [ ] Bump version in CMakeLists.txt, SConstruct, and CHANGELOG (and Python bindings). @MakisH 
+* [ ] Draft message to mailing list @uekerman @MakisH 
+* [ ] Update documentation (all) @MakisH 
+* [ ] Fix potential problems in develop (all)
```
Comment (MakisH) [inline .github/PULL_REQUEST_TEMPLATE/release_pull_request_template.md]:
```suggestion
* [ ] Fix potential problems in develop (all)
* [ ] Approve the PR with at least two reviews (all)
```

File: .github/PULL_REQUEST_TEMPLATE/release_pull_request_template.md [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+* [ ] Merge develop to master @MakisH
```
Comment (MakisH) [inline .github/PULL_REQUEST_TEMPLATE/release_pull_request_template.md]:
```suggestion
* [ ] Merge PR to master @MakisH 
```

File: .github/PULL_REQUEST_TEMPLATE/release_pull_request_template.md [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+* [ ] Bump version in CMakeLists.txt, SConstruct, and CHANGELOG (and Python bindings). @MakisH 
+* [ ] Draft message to mailing list @uekerman @MakisH 
+* [ ] Update documentation (all) @MakisH 
+* [ ] Fix potential problems in develop (all)
+* [ ] Merge develop to master @MakisH 
+* [ ] Tag release (on GitHub) and merge back to develop @MakisH 
+
+## Regression Tests
+
+Run all these tests manually on your system. If you succeed, please write a comment with the revisions of the components that you used below. Example: https://github.com/precice/precice/pull/507#issuecomment-530432289
```
Comment (MakisH) [inline .github/PULL_REQUEST_TEMPLATE/release_pull_request_template.md]: That's a very good note! Apart from the revisions, I would also add more information, as e.g. in [this example](https://github.com/precice/precice/pull/507#issuecomment-530409554).

File: .github/PULL_REQUEST_TEMPLATE/release_pull_request_template.md [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+
+* [ ] SU2/CalculiX @MakisH 
+* [ ] OpenFOAM / OpenFOAM @MakisH 
+* [ ] OpenFOAM / OpenFOAM - NP mapping in OpenFOAM @MakisH
+* [ ] OpenFOAM / OpenFOAM fluid coupling module @MakisH 
+* [ ] OpenFOAM / CalculiX FSI (flap perp) @MakisH
+* [ ] OpenFOAM / CalculiX FSI - NP mapping in CalculiX (3DTube) @KyleDavisSA 
+* [ ] OpenFOAM / CalculiX / OpenFOAM CHT (heat exchanger) @MakisH
+* [ ] OpenFOAM / deal.II @MakisH 
+* [ ] OpenFOAM / FEniCS [flap_perp](https://github.com/precice/tutorials/tree/master/FSI/flap_perp/OpenFOAM-FEniCS) @BenjaminRueth
```
Comment (MakisH) [inline .github/PULL_REQUEST_TEMPLATE/release_pull_request_template.md]: Links are great! Let's add links also in the other cases.

**Comment by MakisH** [comment]
Do I understand correctly that we can only have one PR template and it will always be shown to everybody? We don't want that... :-/ In that case, let's just move it to our `doc/` folder and copy it manually.

**Comment by BenjaminRueth** [comment]
> 
> 
> Do I understand correctly that we can only have one PR template and it will always be shown to everybody? We don't want that... :-/ In that case, let's just move it to our `doc/` folder and copy it manually.

As far as I understand, you can have multiple templates. You can use a link formatted in the way shown above to refer to a specific template (which is not used as the default). And you can provide a default template that will be used for all PRs.

I think if we want to make a new release we should store the link somehere in our docs. Something like [Create a new Release PR](https://github.com/precice/precice/compare/new?template=release_pull_request_template.md).

**Comment by BenjaminRueth** [comment]
This PR is stale for quite a while now. I think we should try to merge it soon, since it will help us with the release.

File: .github/PULL_REQUEST_TEMPLATE/release_pull_request_template.md [hunk 7]
```diff
@@ -0,0 +1,52 @@
+## Step by step guide
+* [ ] Merge master to develop (No commits after the release on master) @MakisH 
+* [ ] Create branch `release-N` from develop. If needed, `git rebase develop`. @MakisH 
+* [ ] Open PR from `release-N` to `master` (use [this template](https://github.com/precice/precice/blob/add_PR_template/.github/PULL_REQUEST_TEMPLATE/release_pull_request_template.md)) @MakisH 
+* [ ] Look over [`CHANGELOG.md`](https://github.com/precice/precice/blob/develop/CHANGELOG.md) and add things if necessary and extract summary (all) @fsimonis 
+* [ ] Do regression tests using the release branch (specific revision) _list below :arrow_down:_ (all)
+* [ ] Bump version in CMakeLists.txt, SConstruct, and CHANGELOG (and Python bindings). @MakisH
```
[the remark this answers is not in the document]
Comment (BenjaminRueth) [reply .github/PULL_REQUEST_TEMPLATE/release_pull_request_template.md]: Added this in 
5d0535b

File: .github/PULL_REQUEST_TEMPLATE/release_pull_request_template.md [hunk 8]
```diff
@@ -0,0 +1,52 @@
+## Step by step guide
+* [ ] Merge master to develop (No commits after the release on master) @MakisH 
+* [ ] Create branch `release-N` from develop. If needed, `git rebase develop`. @MakisH 
+* [ ] Open PR from `release-N` to `master` (use [this template](https://github.com/precice/precice/blob/add_PR_template/.github/PULL_REQUEST_TEMPLATE/release_pull_request_template.md)) @MakisH 
+* [ ] Look over [`CHANGELOG.md`](https://github.com/precice/precice/blob/develop/CHANGELOG.md) and add things if necessary and extract summary (all) @fsimonis
```
[the remark this answers is not in the document]
Comment (BenjaminRueth) [reply .github/PULL_REQUEST_TEMPLATE/release_pull_request_template.md]: Added this in 
f1e50e1

File: .github/PULL_REQUEST_TEMPLATE/release_pull_request_template.md (hunk 6 above)
Comment (BenjaminRueth) [reply .github/PULL_REQUEST_TEMPLATE/release_pull_request_template.md]: I think some links are still missing here.

File: .github/PULL_REQUEST_TEMPLATE/release_pull_request_template.md [hunk 9]
```diff
@@ -0,0 +1,62 @@
...
+* [ ] C Dummy
+* [ ] Fortran Dummy
+* [ ] Fortran 2003 Dummy
+* [ ] Python Dummy
+
+
+## Post-release
+* [ ] Generate packages
+   * [ ] Ubuntu 18.04
+   * [ ] Ubuntu 19.04
```
Comment (MakisH) [inline .github/PULL_REQUEST_TEMPLATE/release_pull_request_template.md]:
```suggestion
   * [ ] Ubuntu 19.10
```

File: .github/PULL_REQUEST_TEMPLATE/release_pull_request_template.md [hunk 10]
```diff
@@ -0,0 +1,62 @@
...
+
+
+## Post-release
+* [ ] Generate packages
+   * [ ] Ubuntu 18.04
+   * [ ] Ubuntu 19.10
+   * [ ] Arch Linux AUR Package
+* [ ] Update Spack recipe
+* [ ] Send email and do marketing
+   * [ ] Mention [xSDK](https://github.com/xsdk-project/xsdk-policy-compatibility/blob/master/precice-policy-compatibility.md)
```
Comment (MakisH) [inline .github/PULL_REQUEST_TEMPLATE/release_pull_request_template.md]:
```suggestion
```
(This was relevant only in v1.6.0)

File: .github/PULL_REQUEST_TEMPLATE/release_pull_request_template.md [hunk 11]
```diff
@@ -0,0 +1,62 @@
...
+* [ ] Tweet
+* [ ] Format the code base
+* [ ] Update the [PR template](https://github.com/precice/precice/blob/add_PR_template/.github/PULL_REQUEST_TEMPLATE/release_pull_request_template.md)?
```
Comment (MakisH) [inline .github/PULL_REQUEST_TEMPLATE/release_pull_request_template.md]: My main question is how do we make sure that we don't always forget how to create a new PR with this template. I suggest adding the following:

```suggestion
* [ ] Update the [PR template](https://github.com/precice/precice/blob/add_PR_template/.github/PULL_REQUEST_TEMPLATE/release_pull_request_template.md)

To open a new PR with this template, use this [PR template query](https://github.com/precice/precice/compare/new?template=release_pull_request_template.md)
```

File: .github/PULL_REQUEST_TEMPLATE/release_pull_request_template.md [hunk 12]
```diff
@@ -0,0 +1,62 @@
+## How to work with this template
+
+* [ ] assign a release manager, who takes care of the process
+* [ ] assign each point below to a responsible person, before you continue. Use `@member`.
```
Comment (MakisH) [inline .github/PULL_REQUEST_TEMPLATE/release_pull_request_template.md]:
```suggestion
* [ ] assign each point below to a responsible person, before you continue. Use `@member`.

Only the release manager should update this post (even tickboxes, due to race conditions in editing). Everybody else should comment on the PR with the progress.
```

File: .github/PULL_REQUEST_TEMPLATE/release_pull_request_template.md [hunk 13]
```diff
@@ -0,0 +1,62 @@
...
+
+## Step by step guide
+* [ ] Merge master to develop (No commits after the release on master)
+* [ ] Create branch `release-N` from develop. If needed, `git rebase develop`.
+* [ ] Open PR from `release-N` to `master` (use [this template](https://github.com/precice/precice/blob/add_PR_template/.github/PULL_REQUEST_TEMPLATE/release_pull_request_template.md))
+* [ ] Look over [`CHANGELOG.md`](https://github.com/precice/precice/blob/develop/CHANGELOG.md) and add things if necessary and extract summary (all)
```
Comment (MakisH) [inline .github/PULL_REQUEST_TEMPLATE/release_pull_request_template.md]:
```suggestion
* [ ] Look over [`CHANGELOG.md`](https://github.com/precice/precice/blob/develop/CHANGELOG.md) (all)
   * Add things, if necessary
   * Extract summary
```

File: .github/PULL_REQUEST_TEMPLATE/release_pull_request_template.md [hunk 14]
```diff
@@ -0,0 +1,62 @@
...
+* [ ] Look over the Roadmap and update entries.
+* [ ] Do regression tests using the release branch (specific revision) _list below :arrow_down:_ (all)
+* [ ] Bump version in:
+   * [ ] [CHANGELOG](https://github.com/precice/precice/blob/develop/CHANGELOG.md)
+   * [ ] [CMakeLists.txt](https://github.com/precice/precice/blob/develop/CMakeLists.txt)
+   * [ ] **???** [Python bindings](https://github.com/precice/python-bindings)
```
Comment (MakisH) [inline .github/PULL_REQUEST_TEMPLATE/release_pull_request_template.md]: Still open.

File: .github/PULL_REQUEST_TEMPLATE/release_pull_request_template.md [hunk 15]
```diff
@@ -0,0 +1,67 @@
...
+   * Add things, if necessary
+   * Extract summary
+* [ ] Look over the Roadmap and update entries.
+* [ ] Do regression tests using the release branch (specific revision) _list below :arrow_down:_ (all)
+* [ ] Bump version in:
+   * [ ] [CHANGELOG](https://github.com/precice/precice/blob/develop/CHANGELOG.md)
+   * [ ] [CMakeLists.txt](https://github.com/precice/precice/blob/develop/CMakeLists.txt)
+   * [ ] **???** [Python bindings](https://github.com/precice/python-bindings) 
+* [ ] Draft message to mailing list
+* [ ] Update documentation (all)
```
Comment (uekerman) [inline .github/PULL_REQUEST_TEMPLATE/release_pull_request_template.md]:
```suggestion
* [ ] Update documentation (all)
  * [ ] Update markdown configuration reference in wiki
```

Comment (uekerman) [reply .github/PULL_REQUEST_TEMPLATE/release_pull_request_template.md]: see #571

File: .github/PULL_REQUEST_TEMPLATE/release_pull_request_template.md (hunk 5 above)
Comment (MakisH) [reply .github/PULL_REQUEST_TEMPLATE/release_pull_request_template.md]: I think that linking to an example is enough to get the idea. Let's not overengineer this.

File: .github/PULL_REQUEST_TEMPLATE/release_pull_request_template.md [hunk 16]
```diff
@@ -0,0 +1,70 @@
...
+* [ ] Tag release (on GitHub, make sure to select the release branch as target) and merge back to develop
+
+## Regression Tests
+
+Run all these tests manually on your system. If you succeed, please write a comment with the revisions of the components that you used below. Example: https://github.com/precice/precice/pull/507#issuecomment-530432289
+
+* [ ] SU2 / CalculiX [flap_perp](https://github.com/precice/tutorials/tree/develop/FSI/flap_perp/SU2-CalculiX)
+* [ ] OpenFOAM / OpenFOAM [flow_over_plate](https://github.com/precice/openfoam-adapter/tree/master/tutorials/CHT/flow-over-plate)
+* [ ] OpenFOAM / OpenFOAM - NP mapping in OpenFOAM [flow_over_plate](https://github.com/precice/openfoam-adapter/tree/master/tutorials/CHT/flow-over-plate)
+* [ ] OpenFOAM / OpenFOAM fluid coupling module **???**
```
Comment (MakisH) [inline .github/PULL_REQUEST_TEMPLATE/release_pull_request_template.md]: This is not really ready for regular testing, it is one of the many under-development cases we have. Let's leave it out for now.
```suggestion
```

File: .github/PULL_REQUEST_TEMPLATE/release_pull_request_template.md [hunk 17]
```diff
@@ -0,0 +1,70 @@
...
+* [ ] Solverdummy [C](https://github.com/precice/precice/tree/develop/tools/solverdummies/c)
+* [ ] Solverdummy [Fortran](https://github.com/precice/precice/tree/develop/tools/solverdummies/fortran)
+* [ ] Solverdummy [Fortran 2003](https://github.com/precice/precice/tree/develop/tools/solverdummies/f2003)
+* [ ] Solverdummy [Python](https://github.com/precice/python-bindings/tree/develop/solverdummy)
+* [ ] Solverdummy [MATLAB](https://github.com/precice/matlab-bindings/tree/develop/solverdummy)
+
+
+## Post-release
+* [ ] Generate packages
+   * [ ] Ubuntu 18.04
```
Comment (MakisH) [inline .github/PULL_REQUEST_TEMPLATE/release_pull_request_template.md]:
```suggestion
   * [ ] Latest Ubuntu LTS
```

File: .github/PULL_REQUEST_TEMPLATE/release_pull_request_template.md [hunk 18]
```diff
@@ -0,0 +1,70 @@
...
+   * [ ] Ubuntu 19.10
```
Comment (MakisH) [inline .github/PULL_REQUEST_TEMPLATE/release_pull_request_template.md]:
```suggestion
   * [ ] Latest Ubuntu
```

File: .github/PULL_REQUEST_TEMPLATE/release_pull_request_template.md [hunk 19]
```diff
@@ -0,0 +1,70 @@
...
+   * [ ] Arch Linux AUR Package
+* [ ] Update Spack recipe
+* [ ] Send email and do marketing
+* [ ] Tweet
+* [ ] Format the code base
```
Comment (MakisH) [inline .github/PULL_REQUEST_TEMPLATE/release_pull_request_template.md]: If we want to keep this in a regular basis, we should move it in the beginning, before testing.

Comment (BenjaminRueth) [reply .github/PULL_REQUEST_TEMPLATE/release_pull_request_template.md]: Fixed https://github.com/precice/precice/pull/509/commits/b3cba80002021c939e09faebe306835cae45c4da

**Pull request merged**